keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/ec2/asset/ClusterHandler.kt

```diff
@@ -0,0 +1,76 @@
+package com.netflix.spinnaker.keel.ec2.asset
+
+import com.netflix.spinnaker.keel.api.Asset
+import com.netflix.spinnaker.keel.api.AssetName
+import com.netflix.spinnaker.keel.api.ec2.Capacity
+import com.netflix.spinnaker.keel.api.ec2.Cluster
+import com.netflix.spinnaker.keel.api.ec2.HealthCheckType
+import com.netflix.spinnaker.keel.api.ec2.InstanceType
+import com.netflix.spinnaker.keel.api.ec2.Metric
+import com.netflix.spinnaker.keel.api.ec2.ScalingProcess
+import com.netflix.spinnaker.keel.api.ec2.SecurityGroup
+import com.netflix.spinnaker.keel.api.ec2.TerminationPolicy
+import com.netflix.spinnaker.keel.clouddriver.CloudDriverCache
+import com.netflix.spinnaker.keel.clouddriver.CloudDriverService
+import com.netflix.spinnaker.keel.ec2.AmazonAssetHandler
+import com.netflix.spinnaker.keel.ec2.CLOUD_PROVIDER
+import com.netflix.spinnaker.keel.orca.OrcaService
+import org.springframework.http.HttpStatus.NOT_FOUND
+import retrofit.RetrofitError
+import java.time.Duration
+
+class ClusterHandler(
+  private val cloudDriverService: CloudDriverService,
+  private val cloudDriverCache: CloudDriverCache,
+  private val orcaService: OrcaService
+) : AmazonAssetHandler<Cluster> {
+  override fun current(spec: Cluster, request: Asset<Cluster>): Cluster? =
+    cloudDriverService.getCluster(spec)
+
+  override fun converge(assetName: AssetName, spec: Cluster) {
+    TODO("not implemented")
+  }
+
+  override fun delete(assetName: AssetName, spec: SecurityGroup) {
+    TODO("not implemented")
+  }
+
+  private fun CloudDriverService.getCluster(spec: Cluster): Cluster? {
+    try {
+      return activeServerGroup(spec.application, spec.accountName, spec.name, spec.region, CLOUD_PROVIDER)
```

Comment:
Gets the current ASG from River and massages it into our format for comparison with desired state